### PR TITLE
scala-runners: init at 0c0b369

### PR DESCRIPTION
**Pull request by hrhino**
This is a script to run multiple versions of scala either by version number or build hash.

###### Motivation for this change

It's a good script and useful for testing with pre-release and non-current versions of Scala.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS <-- don't have one!
   - [ ] other Linux distributions <-- don't have any but can if you like
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)) <-- manually tested this; coursier and jdk are the only upstreams
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after) <-- Total closure size is 390.4Mb but almost all of that is openjdk and coursier
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by puffnfresh — APPROVED** [review]

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3033,6 +3033,12 @@
     githubId = 1436960;
     name = "Christoph Hrdinka";
   };
+  hrhino = {
```
Comment (raboof) [inline maintainers/maintainer-list.nix:3036]: I think the convention is to put this into a separate commit titled `maintainers: add hrhino` (added #96666)

**Review by raboof — APPROVED** [review]
Tested on NixOS, compiling and running a Scala program on different versions indeed works.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (hrhino) [reply maintainers/maintainer-list.nix:3036]: Done; thanks.

Comment (hrhino) [reply maintainers/maintainer-list.nix:3036]: also pushed a newer version because, why not.

File: pkgs/development/compilers/scala-runners/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchFromGitHub, jre, coursier }:
+
+let
```
Comment (raboof) [inline pkgs/development/compilers/scala-runners/default.nix:3]:
```suggestion
let
```

File: pkgs/development/compilers/scala-runners/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchFromGitHub, jre, coursier }:
+
+let 
+  name = "scala-runners";
+  owner = "dwijnand";
+in stdenv.mkDerivation {
+  inherit name;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scala-runners/default.nix:7]:
```suggestion
stdenv.mkDerivation rec {
  name = "scala-runners";
```

File: pkgs/development/compilers/scala-runners/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scala-runners/default.nix:9]: Please put meta at the end.

File: pkgs/development/compilers/scala-runners/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    homepage = "https://github.com/${owner}/${name}";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scala-runners/default.nix:10]:
```suggestion
    homepage = "https://github.com/dwijnand/scala-runners";
```

File: pkgs/development/compilers/scala-runners/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    description = "An alternative implementation of the Scala distribution's runners";
+    license = licenses.asl20;
+    platforms = platforms.all;
+    maintainers = [ maintainers.hrhino ];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scala-runners/default.nix:14]:
```suggestion
    maintainers = with maintainers; [ hrhino ];
```

File: pkgs/development/compilers/scala-runners/default.nix [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  src = fetchFromGitHub {
+    inherit owner;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scala-runners/default.nix:18]:
```suggestion
    owner = "dwijnand";
```

File: pkgs/development/compilers/scala-runners/default.nix [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    repo = name;
+    rev = "0c0b3690b4126d6dfd15024fc759da07434e7b51";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scala-runners/default.nix:20]: Please use a tag

File: pkgs/development/compilers/scala-runners/default.nix [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    mkdir -p $out/bin $out/lib
+    sed -ie "s_\Wcs\W_ ${coursier}/bin/coursier _" scala-runner
+    cp scala-runner $out/lib
+    ln -s $out/lib/scala-runner $out/bin/scala
+    ln -s $out/lib/scala-runner $out/bin/scalac
+    ln -s $out/lib/scala-runner $out/bin/scalap
+    ln -s $out/lib/scala-runner $out/bin/scaladoc
+  '';
+
+
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scala-runners/default.nix:35]:
```suggestion
  installPhase = ''
    mkdir -p $out/bin $out/lib
    sed -ie "s_\Wcs\W_ ${coursier}/bin/coursier _" scala-runner
    cp scala-runner $out/lib
    ln -s $out/lib/scala-runner $out/bin/scala
    ln -s $out/lib/scala-runner $out/bin/scalac
    ln -s $out/lib/scala-runner $out/bin/scalap
    ln -s $out/lib/scala-runner $out/bin/scaladoc
  '';
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/scala-runners/default.nix (hunk 3 above)
Comment (raboof) [reply pkgs/development/compilers/scala-runners/default.nix:7]: Also, instead of having just a `name`, packages should nowadays have a `pname` and a `version` (https://github.com/NixOS/nixpkgs/issues/103997)

File: pkgs/development/compilers/scala-runners/default.nix (hunk 8 above)
Comment (raboof) [reply pkgs/development/compilers/scala-runners/default.nix:20]: There are no tags for this project?

File: pkgs/development/compilers/scala-runners/default.nix (hunk 9 above)
Comment (raboof) [reply pkgs/development/compilers/scala-runners/default.nix:35]: what changed here? just removing the `set -x`? I guess that's a good idea.

Comment (SuperSandro2000) [reply pkgs/development/compilers/scala-runners/default.nix:35]: yeah. not sure why I selected more lines.

File: pkgs/development/compilers/scala-runners/default.nix (hunk 8 above)
Comment (SuperSandro2000) [reply pkgs/development/compilers/scala-runners/default.nix:20]: thats a shame.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (hrhino) [reply maintainers/maintainer-list.nix:3774]: yikes. thanks.

File: pkgs/development/compilers/scala-runners/default.nix (hunk 2 above)
Comment (hrhino) [reply pkgs/development/compilers/scala-runners/default.nix]: yikes. thanks.

File: pkgs/development/compilers/scala-runners/default.nix (hunk 3 above)
Comment (hrhino) [reply pkgs/development/compilers/scala-runners/default.nix]: @raboof should I use (part of) the git tag? I can bother Dale if he wants to push tags but I think he may not care.

File: pkgs/development/compilers/scala-runners/default.nix (hunk 4 above)
Comment (hrhino) [reply pkgs/development/compilers/scala-runners/default.nix]: Done; thanks.

File: pkgs/development/compilers/scala-runners/default.nix (hunk 5 above)
Comment (hrhino) [reply pkgs/development/compilers/scala-runners/default.nix]: Done; thanks.

File: pkgs/development/compilers/scala-runners/default.nix (hunk 3 above)
Comment (raboof) [reply pkgs/development/compilers/scala-runners/default.nix]: there's no clear consensus on how to version packages where upstream doesn't use tags (#100833), using the (short) git tag seems ok to me in this case.

File: pkgs/development/compilers/scala-runners/default.nix [hunk 10]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, lib, fetchFromGitHub, jre, coursier }:
+
+let
+in stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scala-runners/default.nix:4]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/compilers/scala-runners/default.nix [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  name = "scala-runners";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scala-runners/default.nix:5]:
```suggestion
  pname = "scala-runners";
  version = "unstable-2020-02-02"
```

File: pkgs/top-level/all-packages.nix [hunk 12]
```diff
@@ -10851,6 +10851,10 @@ in
   scala_2_13 = callPackage ../development/compilers/scala/2.x.nix { majorVersion = "2.13"; jre = jdk8; };
 
   scala = scala_2_13;
+  scala-runners = callPackage ../development/compilers/scala-runners/default.nix {
+    coursier = callPackage ../development/tools/coursier { jre = jdk8; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:10855]:
```suggestion
    coursier = coursier.override { jre = jdk8; };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/scala-runners/default.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/development/compilers/scala-runners/default.nix]: @hrhino the convention I follow and I think is the best way to deal is

```
pname = "<name of software>";
version = "unstable-${date of the commit in the format YYYY-MM-DD}";
```  

On the other hand I would prefer `version = "<vcs used, say hg or git>-${date}"`

**Comment by AndersonTorres** [comment]
?

**Comment by hrhino** [comment]
#128060 to replace this because apparently force-push then reopen doesn't work. yay, github.
